src/de.rs

```diff
@@ -458,7 +465,9 @@ impl<'de, R: Read<'de>> Deserializer<R> {
 
     fn parse_number(&mut self, positive: bool, significand: u64) -> Result<ParserNumber> {
         Ok(match tri!(self.peek_or_null()) {
+            #[cfg(feature = "floats")]
```

Comment:
Don't we want to immediately error on these two cases if `cfg(not(feature = "floats"))`?
It may make sense to just provide two implementations for `parse_number` (with and without floats)

Reply:
Right. At the moment those cases fall into the fallback implementation which is made for digits only, which cuts the number at the `.` or `e`.